### PR TITLE
feat: new global function for docs releases

### PR DESCRIPTION
**Pull request by ernestojeda**
New changes for edgex-docs release automation. This request came from the Lenny as part of this post-release activities for edgex docs.

Summary of change:

Automation should take place in edgexRelease.groovy as a new function.

Create "release" (kamakura, etc.) branch with these changes, commit and push

delete docs folder.

On the main branch, commit these changes:

Create a PR for each of these:
PR 1:
1. update index.html with current version number to make it the default, i.e. 2.2 (https://github.com/edgexfoundry/edgex-docs/blob/main/docs/index.html#L6)
2. update versions file with next release, i.e. 2.3 - Levski (https://github.com/edgexfoundry/edgex-docs/blob/main/docs/versions.json)

PR 2:
1. Update swagger links to use next version, i.e. 2.3, commit and push change. (https://github.com/edgexfoundry/edgex-docs/search?q=swagger)

Functional Testing:

Dry Run:
https://jenkins.edgexfoundry.org/job/edgexfoundry/job/cd-management/job/PR-228/22/console

Full Run pushing actual branches:
https://jenkins.edgexfoundry.org/job/edgexfoundry/job/cd-management/job/PR-228/21/console

Signed-off-by: Ernesto Ojeda <ernesto.ojeda@intel.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/edgexfoundry/ci-management/blob/main/.github/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Added labels
## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Sandbox Testing
Test Links :

### CODE REVIEW

**Review by bill-mahoney — DISMISSED** [review]
LGTM

File: vars/edgeXReleaseDocs.groovy [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+        if(edgex.isDryRun()) {
+            echo "git push origin ${branch}"
+            echo prCreate
+        } else {
+            sshagent(credentials: [pushCredentials]) {
+                sh "git push origin ${branch}"
+            }
+
+            docker.image('ghcr.io/supportpal/github-gh-cli').inside('--entrypoint=') {
```
Comment (soda480) [inline vars/edgeXReleaseDocs.groovy:186]: nice!  I like the use of gh cli to facilitate pr creation.

File: vars/edgeXReleaseDocs.groovy [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    sh 'git status'
+
+    commitChange("ci: Automated changes for [${branchName}] release")
+
+    // only push the change when we are not doing a DRY_RUN
+    if(!edgex.isDryRun()) {
+        sshagent(credentials: ['edgex-jenkins-ssh']) {
+            sh "git push origin ${branchName}"
+        }
+    }
```
Comment (soda480) [inline vars/edgeXReleaseDocs.groovy:88]: no else for DryRun condition? echo command?

File: vars/edgeXReleaseDocs.groovy [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+        publishVersionChangesPR(releaseInfo)
+        publishSwaggerChangesPR(releaseInfo)
+    }
+}
+
+def publishReleaseBranch(releaseInfo) {
+    def branchName = releaseInfo.releaseName
+
+    sh "git checkout ${releaseInfo.commitId}"
+    sh "git checkout -b ${branchName}"
```
Comment (soda480) [inline vars/edgeXReleaseDocs.groovy:73]: what is your take on combining subsequent `sh` commands? I like the readability of having them separate, but I have read there is a performance hit, your thoughts?

File: src/test/groovy/edgeXReleaseDocsSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+            // Going to skip testing commit change in this function because it is already covered in another test
+            1 * getPipelineMock('sshagent').call(_) >> { _arguments ->
+                assert ['credentials': ['edgex-jenkins-ssh']] == _arguments[0][0]
+            }
+            1 * getPipelineMock('DockerImageMock.inside').call(_) >> { _arguments ->
+                assert "--entrypoint=" == _arguments[0][0]
+            }
+            1 * getPipelineMock('sh').call('git push origin mock-branch')
+            1 * getPipelineMock('sh').call("gh pr create --base main --head mock-branch --title 'mock title' --body 'mock message' --reviewer 'reviwer1,reviwer2' --label 'ci,documentation'")
+    }
```
Comment (soda480) [inline src/test/groovy/edgeXReleaseDocsSpec.groovy:47]: nice, clean unit tests

**Review by soda480 — APPROVED** [review]
Looks very good, I just had a couple of questions and a minor comment.

File: vars/edgeXReleaseDocs.groovy (hunk 3 above)
Comment (ernestojeda) [reply vars/edgeXReleaseDocs.groovy:73]: Performance hit is minor, in my testing everything runs in seconds.

**Review by soda480 — APPROVED** [review]

**Review by bill-mahoney — APPROVED** [review]
LGTM

**Review by bill-mahoney — APPROVED** [review]

**Review by soda480 — APPROVED** [review]

**Pull request merged**